### PR TITLE
Order dev

### PR DESCRIPTION
**Pull request by don-bigdad**

### CODE REVIEW

File: src/main/java/com/example/springbootbookshop/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                                      @PathVariable @Positive Long itemId) {
+        return orderService.getOrderItem(getUserId(authentication),
+                orderId, itemId);
+    }
+
+    @GetMapping("/{orderId}/items")
+    @Operation(summary = "Get special order from user history")
+    @PreAuthorize("hasRole('USER')")
+    public Set<OrderItemDto> showOrder(Authentication authentication,
+                                       @PathVariable Long orderId) {
```
Comment (OlehReex) [inline src/main/java/com/example/springbootbookshop/controller/OrderController.java:69]: you can add  here `@Positive` validation too

File: src/main/java/com/example/springbootbookshop/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+
+    @PostMapping
+    @Operation(summary = "Make order")
+    @PreAuthorize("hasRole('USER')")
+    public CartDto makeOrder(Authentication authentication,
+                             @RequestBody RequestOrderDto requestOrderDto) {
+        return orderService.save(getUserId(authentication), requestOrderDto);
```
Comment (OlehReex) [inline src/main/java/com/example/springbootbookshop/controller/OrderController.java:37]: When you make order you should return response with shippind adress. Create one more DTO smth like `CreateOrderRespponseDto`

File: src/main/java/com/example/springbootbookshop/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @PatchMapping("/{orderId}")
+    @Operation(summary = "Update order status (for admin only)")
+    @PreAuthorize("hasRole('ADMIN')")
+    public void updateStatus(@Positive @PathVariable Long orderId,
+                             @RequestBody UpdateRequestOrderStatus updateStatus) {
+        orderService.updateOrderStatus(orderId, updateStatus);
```
Comment (OlehReex) [inline src/main/java/com/example/springbootbookshop/controller/OrderController.java:45]: Here you should return response with status

File: src/main/java/com/example/springbootbookshop/repository/OrderItemsRepository.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.Optional;
+import java.util.Set;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface OrderItemsRepository extends JpaRepository<OrderItem, Long> {
+    Optional<OrderItemDto> findByOrderAndId(Order order, Long id);
+
+    Set<OrderItemDto> findByOrder(Order order);
```
Comment (OlehReex) [inline src/main/java/com/example/springbootbookshop/repository/OrderItemsRepository.java:15]: Why you return DTO's from database? You should return OrderItem and map it in mapper to OrderItemDto if you need

File: src/main/java/com/example/springbootbookshop/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    private final OrderMapper orderMapper;
+
+    @Override
+    @Transactional
+    public CartDto save(Long userId, RequestOrderDto requestOrderDto) {
+        Cart cart = cartRepository.getCartByUserId(userId).get();
+        if (cart.getCartItems().isEmpty()) {
+            throw new EntityNotFoundException("You cart is empty");
+        }
+        Order order = new Order();
```
Comment (OlehReex) [inline src/main/java/com/example/springbootbookshop/service/impl/OrderServiceImpl.java:48]: it's better to create new method for creating new order (single responsibility)

**Review by OlehReex — APPROVED** [review]
Agree! Just a few comments 😉

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Seems that this PR contains changes for both Shopping Cart and Order. Please split since it is way too big to review

**Pull request closed without merging**